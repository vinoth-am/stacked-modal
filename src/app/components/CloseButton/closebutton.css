.close-button {
  height: 20px;
  width: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px solid var(--dark-shade-grey);
  border-radius: 20px;
  background-color: var(--white);
}
