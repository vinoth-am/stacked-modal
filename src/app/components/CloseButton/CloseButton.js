import React from "react";
import "./closebutton.css";

export const CloseButton = (props) => {
  const { onClick } = props;
  return (
    <button
      data-testid="closeButton"
      className="close-button"
      type="submit"
      onClick={(data) => onClick(data)}
    >
      {props.children}
    </button>
  );
};
