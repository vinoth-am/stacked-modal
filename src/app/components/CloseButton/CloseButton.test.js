import { render, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { CloseButton } from "./CloseButton";

describe("<CloseButton />", () => {
  const onClick = jest.fn();
  it("CloseButton snapshot", () => {
    const { button } = render(<CloseButton onClick={onClick} />);
    expect(button).toMatchSnapshot();
  });

  it("Validate the close button", () => {
    const { getByTestId } = render(<CloseButton onClick={onClick} />);
    const closeButton = getByTestId("closeButton");
    expect(closeButton).toBeTruthy();
  });

  it("Validate the close button text", () => {
    const { getByTestId } = render(
      <CloseButton onClick={onClick}>X</CloseButton>
    );
    const closeButton = getByTestId("closeButton");
    expect(closeButton).toHaveTextContent("X");
  });

  it("Validate the close button click", () => {
    const { getByTestId } = render(<CloseButton onClick={onClick} />);
    const closeButton = getByTestId("closeButton");
    fireEvent.click(closeButton);
    expect(onClick).toBeCalledTimes(1);
  });
});
