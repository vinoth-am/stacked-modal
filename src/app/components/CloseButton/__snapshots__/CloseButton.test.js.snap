// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CloseButton /> CloseButton snapshot 1`] = `undefined`;
